tools/snitch/snitcher/src/client.rs

```diff
@@ -70,9 +71,12 @@ where
         let path_copy = path.to_owned();
 
         let scheme = self.host_name.scheme();
-        let base_path = self.host_name.as_str();
+        let base_path = self.host_name.to_base_path().expect(&format!(
+            "Error when parse base path from {}",
+            self.host_name.as_str()
+        ));
         let path = format!("{}{}", path, query);
-        UrlConnector::build_hyper_uri(scheme, base_path, &path)
+        UrlConnector::build_hyper_uri(scheme, base_path.to_str().unwrap(), &path)
```

Comment:
Why is there an `unwrap` here? Also, consider changing that to an `expect` call.

Reply:
to_str() will return Option<&str>.  will use expect call instead.